Clang: error: equality comparison with extraneous parentheses

# Required information

 * Distribution: Gentoo
 * Distribution version: current
 * The output of "lxc info" or if that fails:
   * Kernel version: N/A
   * LXC version: N/A
   * LXD version: 4.0.1
   * Storage backend in use: N/A
   * Clang version: 10.0

# Issue description

Compile failed under clang

# Steps to reproduce

 1. set CC=clang, CXX=clang++
 2. compile

# Information to attach
```

TERM='dumb' clang -I /var/tmp/portage/app-emulation/lxd-4.0.1/work/lxd-4.0.1/_dist/src/github.com/lxc/lxd/lxd/seccomp -fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=$WORK/b300=/tmp/go-build -gno-record-gcc-switches -I ./ -I/var/tmp/portage/app-emulation/lxd-4.0.1/work/lxd-4.0.1/_dist/deps/sqlite/ -I/var/tmp/portage/app-emulation/lxd-4.0.1/work/lxd-4.0.1/_dist/deps/dqlite/include/ -I/var/tmp/portage/app-emulation/lxd-4.0.1/work/lxd-4.0.1/_dist/deps/raft/include/ -I/var/tmp/portage/app-emulation/lxd-4.0.1/work/lxd-4.0.1/_dist/deps/libco/ -std=gnu11 -Wvla -Werror -fvisibility=hidden -Winit-self -Wformat=2 -Wshadow -Wendif-labels -fasynchronous-unwind-tables -pipe --param=ssp-buffer-size=4 -g -Wunused -Werror=implicit-function-declaration -Werror=return-type -Wendif-labels -Werror=overflow -Wnested-externs -fexceptions -o ./_x003.o -c seccomp.cgo2.c
# github.com/lxc/lxd/lxd/seccomp
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:93:12: error: equality comparison with extraneous parentheses [-Werror,-Wparentheses-equality]
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:93:12: note: remove extraneous parentheses around the comparison to silence this warning
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:93:12: note: use '=' to turn this equality comparison into an assignment
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:95:17: error: equality comparison with extraneous parentheses [-Werror,-Wparentheses-equality]
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:95:17: note: remove extraneous parentheses around the comparison to silence this warning
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:95:17: note: use '=' to turn this equality comparison into an assignment
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:97:17: error: equality comparison with extraneous parentheses [-Werror,-Wparentheses-equality]
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:97:17: note: remove extraneous parentheses around the comparison to silence this warning
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:97:17: note: use '=' to turn this equality comparison into an assignment
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:99:17: error: equality comparison with extraneous parentheses [-Werror,-Wparentheses-equality]
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:99:17: note: remove extraneous parentheses around the comparison to silence this warning
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:99:17: note: use '=' to turn this equality comparison into an assignment
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:101:17: error: equality comparison with extraneous parentheses [-Werror,-Wparentheses-equality]
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:101:17: note: remove extraneous parentheses around the comparison to silence this warning
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:101:17: note: use '=' to turn this equality comparison into an assignment
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:103:17: error: equality comparison with extraneous parentheses [-Werror,-Wparentheses-equality]
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:103:17: note: remove extraneous parentheses around the comparison to silence this warning
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:103:17: note: use '=' to turn this equality comparison into an assignment
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:105:17: error: equality comparison with extraneous parentheses [-Werror,-Wparentheses-equality]
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:105:17: note: remove extraneous parentheses around the comparison to silence this warning
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:105:17: note: use '=' to turn this equality comparison into an assignment
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:107:17: error: equality comparison with extraneous parentheses [-Werror,-Wparentheses-equality]
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:107:17: note: remove extraneous parentheses around the comparison to silence this warning
_dist/src/github.com/lxc/lxd/lxd/seccomp/seccomp.go:107:17: note: use '=' to turn this equality comparison into an assignment

```

--- a/lxd/main_forksyscall.go
+++ b/lxd/main_forksyscall.go
@@ -34,12 +34,6 @@ extern void attach_userns_fd(int ns_fd);
 extern int pidfd_nsfd(int pidfd, pid_t pid);
 extern bool setnsat(int ns_fd, const char *ns);
 
-static inline bool same_fsinfo(struct stat *s1, struct stat *s2,
-			       struct statfs *sfs1, struct statfs *sfs2)
-{
-	return ((sfs1->f_type == sfs2->f_type) && (s1->st_dev == s2->st_dev));
-}
-
 static bool chdirchroot_in_mntns(int cwd_fd, int root_fd)
 {
 	ssize_t len;
@@ -209,8 +203,6 @@ const char *ns_names[] = { "user", "pid", "uts", "ipc", "net", "cgroup", NULL };
 
 static bool change_creds(int ns_fd, cap_t caps, uid_t nsuid, gid_t nsgid, uid_t nsfsuid, gid_t nsfsgid)
 {
-	__do_close int fd = -EBADF;
-
 	if (prctl(PR_SET_KEEPCAPS, 1))
 		return false;
 

--- a/lxd/seccomp/seccomp.go
+++ b/lxd/seccomp/seccomp.go
@@ -91,21 +91,21 @@ static int device_allowed(dev_t dev, mode_t mode)
 {
 	switch (mode & S_IFMT) {
 	case S_IFCHR:
-		if ((dev == makedev(0, 0))) // whiteout
+		if (dev == makedev(0, 0)) // whiteout
 			return 0;
-		else if ((dev == makedev(5, 1))) // /dev/console
+		else if (dev == makedev(5, 1)) // /dev/console
 			return 0;
-		else if ((dev == makedev(1, 7))) // /dev/full
+		else if (dev == makedev(1, 7)) // /dev/full
 			return 0;
-		else if ((dev == makedev(1, 3))) // /dev/null
+		else if (dev == makedev(1, 3)) // /dev/null
 			return 0;
-		else if ((dev == makedev(1, 8))) // /dev/random
+		else if (dev == makedev(1, 8)) // /dev/random
 			return 0;
-		else if ((dev == makedev(5, 0))) // /dev/tty
+		else if (dev == makedev(5, 0)) // /dev/tty
 			return 0;
-		else if ((dev == makedev(1, 9))) // /dev/urandom
+		else if (dev == makedev(1, 9)) // /dev/urandom
 			return 0;
-		else if ((dev == makedev(1, 5))) // /dev/zero
+		else if (dev == makedev(1, 5)) // /dev/zero
 			return 0;
 	}
 

